nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-api/src/main/java/org/nuxeo/ecm/platform/comment/api/CommentManager.java

```diff
@@ -50,28 +79,57 @@
      * @param comment the comment content
      * @param author the comment author
      * @return the comment document model.
+     * @deprecated since 10.3, use {@link #createComment(CoreSession, Comment)} instead.
      */
+    @Deprecated
     DocumentModel createComment(DocumentModel docModel, String comment, String author);
 
+    /**
+     * Creates a comment document model, filling its properties with given info and linking it to given document.
+     *
+     * @param docModel the document to comment
+     * @param comment the comment document model
+     * @return the created comment document model.
+     */
     DocumentModel createComment(DocumentModel docModel, DocumentModel comment);
 
+    /**
+     * Creates a comment document model, filling its properties with given info and linking it to given document.
+     *
+     * @param docModel the document to comment
+     * @param parent the comment parent document model
+     * @param child the comment child document model
+     * @return the created comment document model.
+     * @deprecated since 10.3, use {@link #createComment(CoreSession, Comment)} instead.
+     */
+    @Deprecated
     DocumentModel createComment(DocumentModel docModel, DocumentModel parent, DocumentModel child);
 
+    /**
+     * Deletes a comment.
+     *
+     * @param docModel the comment document model
+     * @param comment the comment
+     * @deprecated since 10.3, use {@link #deleteComment(CoreSession, String)} instead.
+     */
```

Comment:
could we use this important change to order methods by name ? it would be very easier to read, for example, the deleteComment mentionned here is at the bottom of the class

Reply:
A little bit too restrictive IMO